js/modules/k6/http/request.go

```diff
@@ -105,36 +70,49 @@ func (h *HTTP) Request(ctx context.Context, method string, url goja.Value, args
 		params = args[1]
 	}
 
-	req, err := h.parseRequest(ctx, method, url, body, params)
+	req, err := c.parseRequest(method, url, body, params)
 	if err != nil {
 		if state.Options.Throw.Bool {
 			return nil, err
 		}
 		state.Logger.WithField("error", err).Warn("Request Failed")
-		r := httpext.NewResponse(ctx)
+		r := httpext.NewResponse()
 		r.Error = err.Error()
 		var k6e httpext.K6Error
 		if errors.As(err, &k6e) {
 			r.ErrorCode = int(k6e.Code)
 		}
-		return &Response{Response: r}, nil
+		return &Response{Response: r, client: c}, nil
 	}
 
-	resp, err := httpext.MakeRequest(ctx, req)
+	resp, err := httpext.MakeRequest(c.moduleInstance.vu.Context(), state, req)
 	if err != nil {
 		return nil, err
 	}
-	processResponse(ctx, resp, req.ResponseType)
-	return h.responseFromHttpext(resp), nil
+	c.processResponse(resp, req.ResponseType)
+	return c.responseFromHttpext(resp), nil
+}
+
+// processResponse stores the body as an ArrayBuffer if indicated by
+// respType. This is done here instead of in httpext.readResponseBody to avoid
+// a reverse dependency on js/common or goja.
+func (c *Client) processResponse(resp *httpext.Response, respType httpext.ResponseType) {
+	if respType == httpext.ResponseTypeBinary && resp.Body != nil {
+		resp.Body = c.moduleInstance.vu.Runtime().NewArrayBuffer(resp.Body.([]byte))
+	}
+}
+
+func (c *Client) responseFromHttpext(resp *httpext.Response) *Response {
```

Comment:
renaming this with the HTTP acronym in capital letter would make it more readable IMO: `responseFromHTTPext`. If I remember correctly, it's also considered best practice to keep acronyms in capital letters.

Reply:
Resolved by https://github.com/grafana/k6/pull/2226/commits/d87b4d6721568af0d3d6a61ec6272d0ae3f7ccbe